### PR TITLE
Install command and service

### LINKED ISSUE #7389: Install command
Since I cannot register to mautic.org, I will share this feature idea here.

It would be great to have commands added to the `InstallBundle`. This would be particularly useful with Docker images of Mautic as one could build a Docker image and automatically install Mautic and setup its plugins.

For the install command, I think we should :

- [ ] rewrite the `InstallController.php` and export all _actual_ install process to a Service
- [ ] add an `InstallCommand` that could use the previous service the same way the controller does

I will try to write a PR for this if I find the time.

### PR DESCRIPTION
**Pull request by madmath03**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | N
| Related user documentation PR URL | 
| Related developer documentation PR URL | mautic/developer-documentation#122
| Issues addressed (#s or URLs) | Resolves #7389
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Feature to allow command line install of Mautic. The objective is to migrate all the install process of `InstallController` to a service called `InstallService` and add an `InstallCommand` called by `php app/console mautic:install` that would leverage the new service.

Might also be useful to export each install steps into a different service / command in case one would only want to execute specific steps (something like `mautic:install:database`, `mautic:install:admin`, etc...).

This feature was funded by @Monogramm

#### Steps to test this PR:
Two things will need to be tested with this PR: 
1. the new install command(s)
2. the regular install wizard

Mautic install:
1. Load up this PR and user composer to install deps: #7395
```bash
git clone --branch feature/install-command git@github.com:Monogramm/mautic.git mautic_install
cd mautic_install
composer install
```
2. Create an initial `local.php` config file with the database parameters. You can also define the mailer properties and admin properties in `local.php` with the same syntax expected by the command line options (do `php bin/console mautic:install --help` for the list of options)
```php
<?php
// Example local.php to test install (to adapt of course)
$parameters = array(
	// Do not set db_driver and mailer_from_name as they are used to assume Mautic is installed
	'db_host' => 'localhost',
	'db_table_prefix' => null,
	'db_port' => 3306,
	'db_name' => 'mautic',
	'db_user' => 'mautic',
	'db_password' => 'mautic',
	'db_backup_tables' => true,
	'db_backup_prefix' => 'bak_',
	'admin_email' => 'admin@yopmail.com',
	'admin_password' => 'mautic',
	'mailer_transport' => null,
	'mailer_host' => null,
	'mailer_port' => null,
	'mailer_user' => null,
	'mailer_password' => null,
	'mailer_api_key' => null,
	'mailer_encryption' => null,
	'mailer_auth_mode' => null,
);

```
3. Execute `php bin/console mautic:install http://your-mautic-domain.com` (see [documentation](https://github.com/mautic/developer-documentation/pull/122/files#diff-860e3ce2541f17effc8aebd6b8cf606bR55))
```
Mautic Install
==============

Parsing options and arguments...
0 - Checking installation requirements...
Missing optional settings:
  - [0] It is recommended to secure your installation with an SSL certificate (https). Starting February 2020 Google Chrome will stop sending third-party cookies in cross-site requests unless the cookies are secure. Tracking will stop working for Mautic instances running on HTTP. Use HTTPS.
  - [1] The <strong>memory_limit</strong> setting in your PHP configuration is lower than the suggested minimum limit of %min_memory_limit%. Mautic can have performance issues with large datasets without sufficient memory.
Ready to Install!
1 - Creating database...
1.1 - Creating schema...
1.2 - Loading fixtures...
2 - Creating admin user...
3 - Email configuration and final steps...

================
Install complete
================

```
4. Check install worked properly and the install wizard is not triggered
5. Execute `php bin/console mautic:install` again and verify that nothing happens:
```
> php bin/console mautic:install http://your-mautic-domain.com

Mautic already installed
```

Mautic wizard:
1. Load up this PR: #7395
2. Go to your Mautic address and execute install wizard
3. Check install worked properly and the install wizard is not triggered anymore

#### List deprecations along with the new alternative:
None

#### List backwards compatibility breaks:
None (?)

### CODE REVIEW

**Comment by madmath03** [comment]
**Install service is ready for test**

Install controller has been migrated and I have not found issues so far.

Install command is also ready but there is one little bug that I think cannot be fix, at least not with the current process.
When installing for the first time, the `InstallCommand` will properly generate the config in `local.php`, from the command line options, but Doctrine will not reload the entity manager configuration even after a cache clear... so the fixtures fail to be installed on first try (see `InstallService.php` for FIXME comment).
Restarting the install command will work though since the config has been previously generated.

The solution is to generate the `local.php` with the DB config **before** calling the install.
This is automatically done in the official docker image so that might be enough.
If not, I would need to implement a new command that would generate/update the initial config. So install would actually need 2 commands:
```
php app/console mautic:install:setup --db_host ...
php app/console mautic:install:execute http://localhost/mautic --admin_firstname ...
```

Not really a big fan of this so I will wait a review before adding this.
I will add the documentation after that.

**Comment by madmath03** [comment]
@npracht Could you remove the WIP flag and mark this for review please ?

**Comment by luizeof** [comment]
@madmath03 can we pass an sql file for database template?

**Comment by madmath03** [comment]
@luizeof hm... Since my main focus here was to provide a command line to execute the same things that the Web controller did, I'm not really sure how we could do that here.

But for my understanding, what is the use case you are suggesting that for ?

If this is to fix the bug I mentioned in my previous comment, this would not solve the issue but that's probably because my explanation was misleading. When I said  "generate the local.php with the DB config before calling the install", I only meant to generate the `local.php` file with the `db_` properties. You do not need to actually initialize the database, the install command does that really well as long as the properties are loaded from `local.php`.

If you meant this as an additional feature, like I said earlier I would not really know how and when to do it (before default Mautic DB install ? after ? replace default mautic DB install ?). Also, if it's an additional feature, shouldn't this be the subject of an additional PR after this one has been validated ?

**Comment by madmath03** [comment]
Hi @luizeof,

Just pinging you to know more about what you mentioned last time. I would be curious to know your use case about the SQL template, hoping it's not a blocking point.

**Comment by virgilwashere** [comment]
@luizeof,

> @madmath03 can we pass an sql file for database template?

Does `doctrine:database:import` the hard part of what you want?

**Comment by RCheesley** [comment]
@luizeof @npracht is there anything else holding this PR back?

**Comment by npracht** [comment]
Hi @RCheesley, that's too technical for me. It needs tests for sure.

**Comment by npracht** [comment]
Hi there!
It has been decided to not create any extra Mautic 2 minor versions (which means no more features in M2) knowing that Mautic 3 is coming very soon.

**We now want to integrate your contribution in the Mautic 3 roadmap as 3.1.0 candidate.**

## How to do?
1. Check if your feature or enhancement is still missing and relevant in Mautic 3
2. **If Yes**: rebase your PR against the 3.x branch
3. **If No**: please consider closing your PR if you are absolutely sure that the feature has made it into Mautic 3 already

**Please report results by commenting on your PR to make us administration easier.**
______
You can more information on how to do all of that on [this blog post "Getting you PR ready for Mautic 3"](https://www.mautic.org/blog/developer/getting-your-pr-ready-mautic-3).

**Comment by madmath03** [comment]
Hi @npracht 

Tried to rebase for Mautic 3 but the diff was too big and I must have messed up something...
I think I'm gonna close this PR and restart from scratch using staging branch directly

**Comment by madmath03** [comment]
Hi @npracht 

It should be okay now with Mautic 3 staging branch.

**Comment by gramakri** [comment]
I have been trying out this change. I get

```
In CheckStep.php line 87:
                                                            
  Call to a member function getSchemeAndHttpHost() on null  
                                                            
```

In older mautic 2.x, there used to be a `if (!empty($request))` but it's now gone in https://github.com/mautic/mautic/blob/1d73a47acda0442873e9dba6456b79bfbb9ead6d/app/bundles/InstallBundle/Configurator/Step/CheckStep.php#L87 . I guess since we run via CLI, the request object is empty now. There are other places in the same file where the empty request check is removed.

**Comment by madmath03** [comment]
Thanks @gramakri .

I will then take a look `CheckStep.php` and see where the request needs to be secured.

**Comment by gramakri** [comment]
Thanks @madmath03 , If it helps this is how we try to initialize mautic (with your patch) - https://git.cloudron.io/cloudron/mautic-app/-/blob/master/start.sh#L133

**Comment by madmath03** [comment]
It should be better now.

**Comment by gramakri** [comment]
@madmath03 thanks, I will test it out later today.

File: app/bundles/InstallBundle/Configurator/Step/CheckStep.php
Comment (RCheesley) [inline app/bundles/InstallBundle/Configurator/Step/CheckStep.php]: Is there a reason for these being removed?

**Review by RCheesley — CHANGES_REQUESTED** [review]
Hi there, thanks for all the work to update this PR! I noticed a few removals in the code which may be contributing to the reduction in test coverage - I think @kuzmany had this same problem in a few recent PR's. With the test coverage in mind, are you able to write some tests to improve the coverage?

File: app/bundles/InstallBundle/Configurator/Step/CheckStep.php
Comment (kuzmany) [reply app/bundles/InstallBundle/Configurator/Step/CheckStep.php]: But this is not issue.
It's standard feature of CS fixer
It use param **no_superfluous_phpdoc_tags** what based on documentation https://cs.symfony.com/ works properly If remove this comments

@escopecz @alanhartless can you look on it, because for me it's really paint-full and time wasting

Comment (kuzmany) [reply app/bundles/InstallBundle/Configurator/Step/CheckStep.php]: CS fixer make it. 
Option **no_superfluous_phpdoc_tags**  make it. Based on docs https://cs.symfony.com/
> Removes @param, @return and @var tags that don’t provide any useful information.

@alanhartless @escopecz We need know what now. Because it's annoying take it back during each commit, waste a lot of time with it.

**Comment by madmath03** [comment]
> With the test coverage in mind, are you able to write some tests to improve the coverage?

Hi @RCheesley 

I can yes, but I'm not sure about **when** I can find the time to do that...

**Comment by RCheesley** [comment]
Ok no problem, we will just bump to a later release if there aren't tests in time for the release freeze :)

**Comment by madmath03** [comment]
> Ok no problem, we will just bump to a later release if there aren't tests in time for the release freeze :)

What's the deadline to be able to add this to the next release ?

This is something that's been open for more than a year and we would really love to have this as soon as possible to deploy Mautic more easily 😉

File: app/bundles/InstallBundle/Configurator/Step/CheckStep.php
Comment (escopecz) [reply app/bundles/InstallBundle/Configurator/Step/CheckStep.php]: It looks like CS Fixer is not doing this. How can it know what type is `$site_url`?

Comment (escopecz) [reply app/bundles/InstallBundle/Configurator/Step/CheckStep.php]: I returned the var types back but I don't have permission to push it to this PR. @madmath03 did you disable contributions from core members for this PR?

Comment (kuzmany) [reply app/bundles/InstallBundle/Configurator/Step/CheckStep.php]: @escopecz It does. If  I set no_superfluous_phpdoc_tags to false, then CS fixer not remove it. But also not remove it on __construct - see this PR https://github.com/mautic/mautic/pull/8981
Maybe some bug in CS fixer, probably related to PHP 7.4, I asume from 7.4 we could remove it.
Tomas wrote in his blog (https://tomasvotruba.com/blog/2020/07/02/how-to-patch-package-in-vendor-yet-allow-its-updates/)-  But since PHP 7.4, we could drop the @var annotation
I am using PHP 7.2, I've disabled CS fixer pre commit at the moment, but it's really annoying.

**Comment by RCheesley** [comment]
If you take a peek at the [milestones](https://github.com/mautic/mautic/milestones) each of them has the release and freeze dates explained in the text. Features will go into minor releases which would be 3.1, 3.2 - these happen quarterly. The next one is the end of August (3.1) 😬.

File: app/bundles/InstallBundle/Configurator/Step/CheckStep.php
Comment (madmath03) [reply app/bundles/InstallBundle/Configurator/Step/CheckStep.php]: > I returned the var types back but I don't have permission to push it to this PR. @madmath03 did you disable contributions from core members for this PR?

@escopecz I did not do that, it simply is that GitHub forbids PR push from forks owned by an Organization (I made those improvements for my company).

Right now, the only solution is to use GitHub "_code suggestion_" when doing code review.

**Comment by madmath03** [comment]
> The next one is the end of August (3.1) 😬.

@RCheesley I can work with that.
I think I can find some time by the end of the month to implement some tests and improve code coverage.

Thanks for the info on the milestones 😉

**Comment by RCheesley** [comment]
@madmath03 just a gentle nudge, the deadline for the 3.1 freeze is 16th August and we need to have it tested as well by then.

**Comment by madmath03** [comment]
> @madmath03 just a gentle nudge, the deadline for the 3.1 freeze is 16th August and we need to have it tested as well by then.

Hi @RCheesley,

I've just added some basic tests to this PR. It does not cover the complete / core install process, but at least improves the coverage on the new code to reach the target.

I hope it will be alright for 3.1 release.

**Comment by madmath03** [comment]
By the way, I think that the "_-5.55%_" on code coverage is due to the fact that the branch is a bit outdated compared to the staging branch since all files mentioned are not impacted by this PR.

**Comment by madmath03** [comment]
hi @RCheesley ,

Just wanted to notify that I've did some corrections after a few tests on my side.

PR is ready for review / test and hopefully merge, so just want to make sure this is not missed for next release.

PS: Codecov seems to be stuck somehow...

**Comment by RCheesley** [comment]
@madmath03 you need to rebase this PR against the Staging branch - that should fix the code coverage issue :) Shout if you need a hand with that!

**Comment by madmath03** [comment]
> you need to rebase this PR against the Staging branch

@RCheesley Done

**Comment by RCheesley** [comment]
@madmath03 it looks like the codecov job got stuck .. can you try another rebase/force push? I tried but I don't have permission on your repo. Will take a look at testing today.

**Comment by RCheesley** [comment]
@madmath03 I'm trying to test this following the instructions.

I've updated them to use bin/console now we're on Mautic 3.x where the location is different.

I've also explained (based on your documentation) that you need to specify the install URL in the command. The documentation doesn't make reference to any parameters needing to be passed - is this a requirement or does it now pull that from the local.php file?

Here's what I've found:

local.php set up using one from my Mautic 3.0.2 instance, changing the database and site URL.

Run the command as instructed:

```
ruth.cheesley@MacBookPro-RuthCheesley mautic % bin/console mautic:install http://local.m3testinstall
                                                                                                                        
 [WARNING] Some commands could not be registered:                                                                       
                                                                                                                        
In DateHelper.php line 47:
                                                                                                                                                                                                         
  Type error: Too few arguments to function Mautic\CoreBundle\Templating\Helper\DateHelper::__construct(), 5 passed in /Users/ruth.cheesley/Sites/local.m3testinstall/mautic/var/cache/prod/ContainerZp  
  eqf7x/getMautic_Helper_Template_DateService.php on line 8 and exactly 6 expected                                                                                                                       
                                                                                                                                                                                                         
In DateHelper.php line 47:
                                                                                                                                                                                                         
  Type error: Too few arguments to function Mautic\CoreBundle\Templating\Helper\DateHelper::__construct(), 5 passed in /Users/ruth.cheesley/Sites/local.m3testinstall/mautic/var/cache/prod/ContainerZp  
  eqf7x/getMautic_Helper_Template_DateService.php on line 8 and exactly 6 expected                                                                                                                       
                                                                                                                                                                                                         
In DateHelper.php line 47:
                                                                                                                                                                                                         
  Type error: Too few arguments to function Mautic\CoreBundle\Templating\Helper\DateHelper::__construct(), 5 passed in /Users/ruth.cheesley/Sites/local.m3testinstall/mautic/var/cache/prod/ContainerZp  
  eqf7x/getMautic_Helper_Template_DateService.php on line 8 and exactly 6 expected                                                                                                                       
                                                                                                                                                                                                         
In DateHelper.php line 47:
                                                                                                                                                                                                         
  Type error: Too few arguments to function Mautic\CoreBundle\Templating\Helper\DateHelper::__construct(), 5 passed in /Users/ruth.cheesley/Sites/local.m3testinstall/mautic/var/cache/prod/ContainerZp  
  eqf7x/getMautic_Helper_Template_DateService.php on line 8 and exactly 6 expected                                                                                                                       
                                                                                                                                                                                                         
In DateHelper.php line 47:
                                                                                                                                                                                                         
  Type error: Too few arguments to function Mautic\CoreBundle\Templating\Helper\DateHelper::__construct(), 5 passed in /Users/ruth.cheesley/Sites/local.m3testinstall/mautic/var/cache/prod/ContainerZp  
  eqf7x/getMautic_Helper_Template_DateService.php on line 8 and exactly 6 expected                                                                                                                       
                                                                                                                                                                                                         
In Container.php line 348:
                                                                                                                  
  You have requested a non-existent service "mautic.install.service". Did you mean this: "mautic.queue.service"?  
                                                                                                                  
mautic:install [-f|--force] [--db_driver DB_DRIVER] [--db_host DB_HOST] [--db_port DB_PORT] [--db_name DB_NAME] [--db_user DB_USER] [--db_password DB_PASSWORD] [--db_table_prefix DB_TABLE_PREFIX] [--db_backup_tables DB_BACKUP_TABLES] [--db_backup_prefix DB_BACKUP_PREFIX] [--admin_firstname ADMIN_FIRSTNAME] [--admin_lastname ADMIN_LASTNAME] [--admin_username ADMIN_USERNAME] [--admin_email ADMIN_EMAIL] [--admin_password ADMIN_PASSWORD] [--mailer_from_name [MAILER_FROM_NAME]] [--mailer_from_email [MAILER_FROM_EMAIL]] [--mailer_transport [MAILER_TRANSPORT]] [--mailer_host MAILER_HOST] [--mailer_port MAILER_PORT] [--mailer_user MAILER_USER] [--mailer_password [MAILER_PASSWORD]] [--mailer_encryption [MAILER_ENCRYPTION]] [--mailer_auth_mode [MAILER_AUTH_MODE]] [--mailer_spool_type MAILER_SPOOL_TYPE] [--mailer_spool_path MAILER_SPOOL_PATH] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--] <command> <site_url> [<step>]

```

**Comment by madmath03** [comment]
> @madmath03 it looks like the codecov job got stuck .. can you try another rebase/force push?

Sure! it's done. Let's hope it's not going to get stuck again 😅 

> I tried but I don't have permission on your repo. Will take a look at testing today.

I made this enhancement as part of my work, thus stored in my company's organization, but GitHub does not allow push forks from organizations...

**Comment by RCheesley** [comment]
Fab, thanks so much and yes no problem - we come up against that with a few orgs who contribute, it just means that we're dependant on the contributor rebasing etc if needed.

Thanks for getting back to us so quickly!

Any ideas re. the error I came across? Did I miss a step somewhere?

**Comment by madmath03** [comment]
> I've updated them to use bin/console now we're on Mautic 3.x where the location is different.

Thanks for mentioning that, I updated the documentation PR accordingly: mautic/developer-documentation#122

> I've also explained (based on your documentation) that you need to specify the install URL in the command. The documentation doesn't make reference to any parameters needing to be passed - is this a requirement or does it now pull that from the local.php file?

The site URL is required, but the other parameters are optional and can be retrieved from `local.php` (that's going to be very useful when using the docker image which initializes the file from env var)...

> Any ideas re. the error I came across? Did I miss a step somewhere?

@RCheesley ...well, I've tested this command last week with the DB parameters in command line (on Windows) without any issues, so I'm confused by this error.
Your install command seems to be correct, but more importantly it seems like your error is not related at all with the command itself but more to some setup or code issues.

Could it be possible that your cache contains information from another branch ?

I will try to test it again by deleting my DB, clone the branch and try install with `local.php` initialized instead of using command line.

**Comment by RCheesley** [comment]
I'll give it another test and see if I can get it to work! This was on an instance I have for testing installations which I don't believe had any other PR's or code applied (I always start on staging before pulling the PR) but I'll pull down and start from fresh just to be sure!

**Review by RCheesley — APPROVED** [review]
Thanks @madmath03 for your help with testing this PR! I got it to work in the end, the only part I was a little concerned by, but it didn't seem to affect the install, was the following log entries during the install:

```
[2020-08-15 20:27:22] console.ERROR: Error thrown while running command "doctrine:migrations:version --add --all --no-interaction". Message: "Migration version 20181111095447 already registered with class Doctrine\Migrations\Version\Version" {"exception":"[object] (Doctrine\\Migrations\\Exception\\DuplicateMigrationVersion(code: 7): Migration version 20181111095447 already registered with class Doctrine\\Migrations\\Version\\Version at /Users/ruth.cheesley/Sites/local.m3testinstall/mautic/vendor/doctrine/migrations/lib/Doctrine/Migrations/Exception/DuplicateMigrationVersion.php:14)","command":"doctrine:migrations:version --add --all --no-interaction","message":"Migration version 20181111095447 already registered with class Doctrine\\Migrations\\Version\\Version"} []
[2020-08-15 20:27:23] console.ERROR: Error thrown while running command "doctrine:migrations:version --add --all --no-interaction". Message: "Migration version 20181111095447 already registered with class Doctrine\Migrations\Version\Version" {"exception":"[object] (Doctrine\\Migrations\\Exception\\DuplicateMigrationVersion(code: 7): Migration version 20181111095447 already registered with class Doctrine\\Migrations\\Version\\Version at /Users/ruth.cheesley/Sites/local.m3testinstall/mautic/vendor/doctrine/migrations/lib/Doctrine/Migrations/Exception/DuplicateMigrationVersion.php:14)","command":"doctrine:migrations:version --add --all --no-interaction","message":"Migration version 20181111095447 already registered with class Doctrine\\Migrations\\Version\\Version"} []
[2020-08-15 20:27:23] console.ERROR: Error thrown while running command "doctrine:migrations:version --add --all --no-interaction". Message: "Migration version 20181111095447 already registered with class Doctrine\Migrations\Version\Version" {"exception":"[object] (Doctrine\\Migrations\\Exception\\DuplicateMigrationVersion(code: 7): Migration version 20181111095447 already registered with class Doctrine\\Migrations\\Version\\Version at /Users/ruth.cheesley/Sites/local.m3testinstall/mautic/vendor/doctrine/migrations/lib/Doctrine/Migrations/Exception/DuplicateMigrationVersion.php:14)","command":"doctrine:migrations:version --add --all --no-interaction","message":"Migration version 20181111095447 already registered with class Doctrine\\Migrations\\Version\\Version"} []
[2020-08-15 20:27:23] console.ERROR: Error thrown while running command "doctrine:migrations:version --add --all --no-interaction". Message: "Migration version 20181111095447 already registered with class Doctrine\Migrations\Version\Version" {"exception":"[object] (Doctrine\\Migrations\\Exception\\DuplicateMigrationVersion(code: 7): Migration version 20181111095447 already registered with class Doctrine\\Migrations\\Version\\Version at /Users/ruth.cheesley/Sites/local.m3testinstall/mautic/vendor/doctrine/migrations/lib/Doctrine/Migrations/Exception/DuplicateMigrationVersion.php:14)","command":"doctrine:migrations:version --add --all --no-interaction","message":"Migration version 20181111095447 already registered with class Doctrine\\Migrations\\Version\\Version"} []
```

I double checked and there were no outstanding migrations, so it seems like they were applied correctly.

I checked the automated install, and also the install via the regular GUI, with this PR installed. 🎉 🚀

**Comment by madmath03** [comment]
You're welcome @RCheesley 😃 

I got the same logs on my side, but detected nothing wrong.
I'm guessing the doctrine steps tries to execute the same migrations several times, but I did not find any differences between the InstallController and InstallCommand process and both delegate install to InstallService...

If any Mautic core team dev can check, that would be cool.

**Comment by RCheesley** [comment]
@mautic/core-team if someone could have a glance over, it would be appreciated 👍

**Review by dennisameling — APPROVED** [review]
The doctrine "error" should be no problem as it's simply indicating that the DB migration is already in place. I just tested and can login to Mautic without problems after running `mautic:install`.

Thanks a lot for this great feature @madmath03!! 🚀

**Comment by rez0n** [comment]
Hi, thanks for the great tool.
I've thought it is more flexible if the install script accepts default (unspecified) MySQL port.
This issue according to the Mautic docker, if the `MAUTIC_DB_PORT` environment variable was omitted Mautic use default 3306 port, but install script won't.

```
========================================================================
Mautic Install
==============

Parsing options and arguments...
0 - Checking installation requirements...
Missing optional settings:
  - [0] It is recommended to secure your installation with an SSL certificate (https). Starting February 2020 Google Chrome will stop sending third-party cookies in cross-site requests unless the cookies are secure. Tracking will stop working for Mautic instances running on HTTP. Use HTTPS.
Ready to Install!
1 - Creating database...
Errors in database configuration/installation:
  - [port] The port must be an integer greater than 0.
Install canceled
```

**Pull request merged**
